src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -96,6 +96,15 @@ void handleBCProp(HandlerContext& handlerContext)
     }
 }
 
+void handleSource(HandlerContext& handlerContext)
+{
+    auto new_source = Source();
+    for (const auto& record : handlerContext.keyword) {
+        new_source.updateSource(record);
+    }
+    handlerContext.state().source.update(std::move(new_source));
```

Comment:
This definition (especially `new_source = Source{}`) will replace all existing source terms with the contents of the new `SOURCE` keyword if the keyword is entered multiple times.  Is that the intended/desired semantics of this keyword?

Reply:
That was not intended. The intention (also reflected in one of the test cases) was that old source terms should be kept until they are overwritten by a source with the same i.j,k index. Strangely the test pass. 

```
SCHEDULE
SOURCE
 1 1 1 GAS 0.01 /
 1 1 1 WATER 0.01 /
/
DATES             -- 1
 10  'JUN'  2007 /
/
SOURCE
 1 1 1 GAS 0.0 /
 1 1 2 WATER 0.02 /
/
```
Still gives 0.01 of water for cell 1 1 1.

Reply:
I have updated the code and added a test for Schedule. Thanks for noticing.